svgdigitizer/electrochemistry/cv.py

```diff
@@ -312,13 +316,124 @@ def x_label(self):
             match[0], match[1], match[2] or "unknown"
         )
 
+    @property
+    @cache
+    def figure_label(self):
+        r"""
+        An identifier of the plot to distinguish it from other figures on the same page.
+
+        The figure name is read from a ``<text>`` in the SVG file
+        such as ``<text>figure: 2b</text>``.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from svgdigitizer.svgplot import SVGPlot
+            >>> from svgdigitizer.electrochemistry.cv import CV
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">x1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">x2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">y1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">y2: 1 A</text>
+            ...   </g>
+            ...   <text x="-200" y="330">Figure: 2b</text>
+            ... </svg>'''))
+            >>> cv = CV(SVGPlot(svg))
+            >>> cv.figure_label
+            '2b'
+
+        """
+        figure_labels = self.svgplot.svg.get_texts("(?:figure): (?P<label>.+)")
+
+        if len(figure_labels) > 1:
+            logger.warning(
+                f"More than one text field with figure labels. Ignoring all text fields except for the first: {figure_labels[0]}."
+            )
+
+        if not figure_labels:
```

Comment:
I rewrote all the properties so they fall back to the yaml metadata when no value has been specified in the SVG. What do you think?

Reply:
Sure we can do it like that. For the reviewer of the files, it might be a bit awkward since one has to check both the SVG and the YAML to figure out if something is missing.
We can always revoke this change if we find out later that it is not convenient.

Reply:
Sure. Then again, we should make sure that the checks in the website repository alert us automatically if something is missing.